### PR TITLE
docs: states page

### PR DESCRIPTION
**Pull request by anna-geller**
closes https://github.com/kestra-io/kestra/issues/2556

### CODE REVIEW

**Comment by wrussell1999** [comment]
@anna-geller there's a load of info on State under the execution page. Would it make more sense to add to that rather than create a separate page? https://kestra.io/docs/workflow-components/execution#state

File: content/docs/04.workflow-components/17.states.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+
+## Execution States
+
+Each Kestra execution can transition through several states during its lifecycle. The following diagram illustrates the possible states an execution can be in:
+
+![execution_states](/docs/concepts/execution_states.png)
+
+Here is a brief description of each state:
+1. **CREATED**: The execution has been created but not yet started. This transient state indicates that the execution is waiting to be processed, and it should usually quickly transition to the `RUNNING`, `CANCELLED` or `QUEUED` state. If you see executions stuck in this state, it may indicate a problem with the system.
```
[the remark this answers is not in the document]
Comment (anna-geller) [reply content/docs/04.workflow-components/17.states.md:20]: Yes but note that I added: if executions are stuck in this state i.e. they don't move to RUNNING. We had an issue on the Kafka side in the past that caused hundreds of queued executions. Do you have any suggestions on how to better formulate it?

File: content/docs/04.workflow-components/17.states.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+10. **RESTARTED**: This transient state is equivalent to the `CREATED` state, but for a failed execution that has been restarted e.g. from the UI. Such execution will transition to the `RUNNING` state as soon as the restarted execution is processed.
+11. **CANCELLED**: This terminal state indicates that the execution has been automatically cancelled by the system, usually because the `concurrency` limit was reached and the [concurrency](14.concurrency.md) `bahavior` was set to `CANCEL` in order to cancel all executions that exceed the concurrency limit.
+12. **KILLING**: This transient state indicates that the user has issued a command to kill the execution, e.g. by clicking on the `Kill` button in the UI. The system is in the process of terminating (aka _killing_) the task runs associated with the execution which are still in progress. As soon as all task runs are terminated, the execution will transition to the `KILLED` state.
```
[the remark this answers is not in the document]
Comment (anna-geller) [reply content/docs/04.workflow-components/17.states.md:31]: is it already implemented? didn't find it here https://kestra.io/plugins/core

[the remark this answers is not in the document]
Comment (loicmathieu) [reply content/docs/04.workflow-components/17.states.md]: You're right, it was delayed https://github.com/kestra-io/kestra/issues/3352

**Pull request merged**